### PR TITLE
Create add review

### PR DESCRIPTION
**Pull request by AFahgan**

### CODE REVIEW

File: server/utils/validation/addReviewSchema.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Joi from 'joi';
+
+const addReviewSchema = Joi.object({
+  gymId: Joi.number().required(),
+  description: Joi.string().min(2).max(1000).required(),
```
Comment (LinaYahya) [inline server/utils/validation/addReviewSchema.ts:5]:
```suggestion
  description: Joi.string().min(2).max(1000).allow(""),

```

File: test/addReview.test.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  const newSubscription = {
+    gymId: 1,
+    username: 'ali',
+    userPhone: '0592157001',
+    type: 'month',
+  };
+
+  beforeAll(() => buildDB());
+  describe('Revieq Gyms API Testing', () => {
+    test('POST: Test route Create new subscription ~~ path ==> /api/v1/subscriptions - Success', async () => {
```
Comment (LinaYahya) [inline test/addReview.test.ts:31]: test at least one success review added

File: server/controllers/gyms/addReview.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      gymId,
+      username,
+      rate,
+      description,
+      userPhone,
+    } = await addReviewSchema.validateAsync(req.body, { abortEarly: false });
+    const review = await Review.create({
+      gymId,
+      username,
+      rate,
```
Comment (LinaYahya) [inline server/controllers/gyms/addReview.ts:17]: why you're creating a review before check if subscription valid or not

File: server/controllers/gyms/addReview.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      description,
+    });
+    const isExist: any = await Subscription.findOne({
```
Comment (LinaYahya) [inline server/controllers/gyms/addReview.ts:20]: create an interface for Subscription model to fix types here

File: server/controllers/gyms/addReview.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      where: { userPhone, gymId },
+    });
+    if (!isExist) {
+      throw new CustomError('عذراً لا يوجد لديك إشتراك في هذا النادي لتممكن من إضافة تقييم', 401);
+    }
+    const isActive: any = isExist.status;
```
Comment (LinaYahya) [inline server/controllers/gyms/addReview.ts:26]:
```suggestion
    const { status }: boolean = isExist;

```

File: server/routes/gyms.ts [hunk 6]
```diff
@@ -12,6 +12,7 @@ gyms.post('/register', gymRegister);
 gyms.get('/top', getTopGyms);
 gyms.get('/filter', getFilteredGyms);
 gyms.get('/search', searchGymByName);
+gyms.post('/reviews', addReview);
```
Comment (LinaYahya) [inline server/routes/gyms.ts:15]: better to remove to `gym` router
```suggestion
gym.post('/review', addReview);
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by rawaneyad — APPROVED** [review]
Good Job

**Pull request merged**